topdown/reachable_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package topdown
+
+import (
+	"testing"
+)
+
+func TestReachable(t *testing.T) {
+	data := map[string]interface{}{}
+	tests := []struct {
```

Comment:
Thoughts on adding some negative test cases? I'm thinking cases for like for passing in a non-valid type (eg arg[0] is a string or int or something instead of an object) as well as object values that are not a set/array where we should just skip over them.